src/main/java/Deadline.java

```diff
@@ -0,0 +1,17 @@
+public class Deadline extends Task{
+    private String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D][" + getStatusIcon() + "] " + description + " (by: " + by + ")";
+    }
+    @Override
+    public String toFileString() {
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + by;
```

Comment:
Consider adding the description in the parent classes `toFIleString()` to make full use of the inheritance principles